src/curl_async.c

```diff
@@ -86,13 +86,15 @@ struct swupd_curl_parallel_handle {
 	bool resume_failed;
 	int last_retry; /* keep the largest retry number so far */
 
-	CURLM *mcurl;			  /* Curl handle */
-	struct list *failed;		  /* List of failed downloads */
-	struct hashmap *curl_hashmap;     /* Hashmap mentioned above */
-	struct tp *thpool;		  /* Pointer to the threadpool */
-	swupd_curl_success_cb success_cb; /* Callback to success function */
-	swupd_curl_error_cb error_cb;     /* Callback to error function */
-	swupd_curl_free_cb free_cb;       /* Callback to free user data*/
+	CURLM *mcurl;			    /* Curl handle */
+	struct list *failed;		    /* List of failed downloads */
+	struct hashmap *curl_hashmap;       /* Hashmap mentioned above */
+	struct tp *thpool;		    /* Pointer to the threadpool */
+	swupd_curl_success_cb success_cb;   /* Callback to success function */
+	swupd_curl_error_cb error_cb;       /* Callback to error function */
+	swupd_curl_free_cb free_cb;	 /* Callback to free user data */
+	swupd_curl_progress_cb progress_cb; /* Callback to report download progress */
+	struct download_progress *download_progress;
```

Comment:
You don't use this struct in curl_async. It's more generic to have a void *data here, so we can the user's of the API can define what they need to use.

Reply:
will do that